src/Manager.php

```diff
@@ -40,6 +50,20 @@ class Manager
     public function __construct(RepositoryInterface $repository)
     {
         $this->_repository = $repository;
+        $this->_filterLocator = new FilterLocator($this);
```

Comment:
is it possible to extract this into a class var (static or otherwise) similar to what is done in core?
This would allow at least basic extension here - and sounds cleaner in terms of SOLID (especially since we dont use factory pattern here at all).

Reply:
Can also make the locator injectable through contructor and create an instance only if relevant argument is `null`.

Reply:
Sure, both is fine.

Reply:
@dereuromark are you going to make this change?

Reply:
I guess with the existing setFilterLocator() setter this is probably no big deal here now.